File: frontend/src/stores/document.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import { documentApi } from '../api'

export interface Document {
  id: number
  project: number
  title: string
  file: string
  file_size: number
  file_type: string
  raw_text: string
  parsed_structure: object
  status: string
  created_at: string
}

export const useDocumentStore = defineStore('document', () => {
  const documents = ref<Document[]>([])
  const uploading = ref(false)
  const uploadProgress = ref(0)

  async function fetchDocuments(projectId: number) {
    const { data } = await documentApi.list(projectId)
    documents.value = data
  }

  async function uploadDocument(projectId: number, file: File) {
    uploading.value = true
    uploadProgress.value = 0
    try {
      await documentApi.upload(projectId, file, (p) => { uploadProgress.value = p })
      await fetchDocuments(projectId)
    } finally {
      uploading.value = false
    }
  }

  async function deleteDocument(id: number, projectId: number) {
    await documentApi.delete(id)
    await fetchDocuments(projectId)
  }

  return { documents, uploading, uploadProgress, fetchDocuments, uploadDocument, deleteDocument }
})

File: backend/models/document.py
from datetime import datetime

from sqlalchemy import String, Text, Integer, DateTime, ForeignKey, JSON, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from database import Base


class Document(Base):
    __tablename__ = 'documents_document'

    id: Mapped[int] = mapped_column(primary_key=True)
    project_id: Mapped[int] = mapped_column(ForeignKey('projects_project.id'))
    title: Mapped[str] = mapped_column(String(500))
    file_path: Mapped[str] = mapped_column('file', String(500), default='')
    file_size: Mapped[int] = mapped_column(Integer, default=0)
    file_type: Mapped[str] = mapped_column(String(10), default='')
    raw_text: Mapped[str] = mapped_column(Text, default='')
    parsed_structure: Mapped[dict] = mapped_column(JSON, default=dict)
    status: Mapped[str] = mapped_column(String(20), default='uploaded')
    created_at: Mapped[datetime] = mapped_column(DateTime, default=func.now())

    project: Mapped['Project'] = relationship(back_populates='documents')
    testdesigns: Mapped[list['TestDesign']] = relationship(back_populates='document', cascade='all, delete-orphan')
